aidial_adapter_openai/utils/storage.py

```diff
@@ -51,6 +52,15 @@ async def _get_bucket(self, session: aiohttp.ClientSession) -> Bucket:
 
         return self.bucket
 
+    async def _get_user_bucket(self, session: aiohttp.ClientSession) -> str:
+        bucket = await self._get_bucket(session)
+        appdata = bucket.get("appdata")
```

Comment:
But why we don't use `bucket` field from the [API](https://epam-rail.com/dial_api#tag/Files/paths/~1v1~1bucket/get) response?

Due to schema, it's required, on contrary to `appdata`

Reply:
Because it's two different things:

1. `bucket` - is a bucket of the **application**. It identifies its file space.
2. `appdata` - points to the **user**'s file space, in particular, it points to a folder which the application has the rights to read from and write to.

E.g.
```json
{
    "bucket": "<application_bucket>",
    "appdata": "<john_smith_bucket>/appdata/<application_deployment_id>"
}
```

When the application recieves DIAL URL to a file, it points to the **user**'s file space:

```json 
{
    "url": "files/<john_smith_bucket>/foo/bar/document.pdf"
}
```

So we need to use `appdata` to extract the user's bucket and remove it from the attachment url to make a readable filename without any redundant information.